File: WatchShop/src/java/model/Category.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package model;

/**
 *
 * @author admin
 */
public class Category {

    private String cid;
    private String cname;
    private String type;
    private String deleted;

    public Category() {
    }

    public Category(String cid, String cname, String type, String deleted) {
        this.cid = cid;
        this.cname = cname;
        this.type = type;
        this.deleted = deleted;
    }

    public String getCid() {
        return cid;
    }

    public void setCid(String cid) {
        this.cid = cid;
    }

    public String getCname() {
        return cname;
    }

    public void setCname(String cname) {
        this.cname = cname;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDeleted() {
        return deleted;
    }

    public void setDeleted(String deleted) {
        this.deleted = deleted;
    }

}

File: WatchShop/web/js/category.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */

function validateCategoryName() {
    var categoryName = $("#cname").val();
    if (categoryName.trim() === '') {
        $("#categoryNameError").text("Vui lòng nhập dòng sản phẩm.");
    } else if (categoryName.trim().length < 3) {
        $("#categoryNameError").text("Dòng sản phẩm phải chứa nhiều hơn 3 kí tự.");
    } else {
        $("#categoryNameError").text("");
    }
}

function checkWhitespace() {
    var cname = document.getElementById("cname").value;

    var whitespaceMessageElement = document.getElementById("whitespaceMessage");

    if (/^\s/.test(cname)) {
        whitespaceMessageElement.innerHTML = "Trường không thể bắt đầu bằng dấu cách.";
        whitespaceMessageElement.style.color = "red";
    } else {
        whitespaceMessageElement.innerHTML = "";
    }
}

function submitAddCategoryForm() {
    validateCategoryName();
    checkWhitespace();

    if ($("#categoryNameError").text() || $("#whitespaceMessage").text()) {
        return;
    } else {
        $("#addCategoryForm").submit();
    }
}

$(document).ready(function () {
    $("#cname").blur(validateCategoryName);

    $('.addbtn').on('click', function () {
        $('#addmodal').modal('show');
    });

    $('.editbtn').on('click', function () {
        $('#editmodal').modal('show');
        $tr = $(this).closest('tr');
        var data = $tr.children("td").map(function () {
            return $(this).text();
        }).get();
        $('#update_id').val(data[0].trim());
        $('#update_name').val(data[1].trim());
        $('#update_type').val(data[2].trim());
    });

    $('.deletebtn').on('click', function () {
        var cid = $(this).data('id');
        if (confirm('Bạn có muốn xóa thể loại này ?')) {
            window.location = 'deletecategory?cid=' + cid;
        }
    });

    $('#search-btn').on('click', function (e) {
        e.preventDefault();
        searchCategories();
    });

    $('#search-text').on('keyup', function () {
        searchCategories();
    });
});

function searchCategories() {
    var searchText = $("#search-text").val().toLowerCase();
    var rows = document.querySelectorAll("tbody tr");
    rows.forEach(row => {
        var categoryName = row.children[1].textContent.toLowerCase();
        if (categoryName.includes(searchText)) {
            row.style.display = "";
        } else {
            row.style.display = "none";
        }
    });
}
